napari/layers/shapes/shapes.py

```diff
@@ -2642,18 +2643,30 @@ def _finish_drawing(self, event=None) -> None:
                     self._data_view.edit(index, vertices[:-1])
             if self._mode in {Mode.ADD_POLYGON, Mode.ADD_POLYGON_LASSO}:
                 vertices = self._data_view.shapes[index].data
+                if self._mode == Mode.ADD_POLYGON_LASSO:
+                    prev_vertices = len(vertices)
+                    vertices = rdp(
+                        vertices,
+                        epsilon=get_settings().experimental.rdp_epsilon,
+                    )
+                    if len(vertices) <= 3 and prev_vertices > 3:
+                        eps = get_settings().experimental.rdp_epsilon
+                        removed_vertices = prev_vertices - len(vertices)
+                        show_warning(
+                            trans._(
+                                'Polygon must have at least 3 vertices. With current rep epsilon value {eps},'
+                                ' the {removed_vertices} vertices were removed.',
+                                'Cannot create polygon.',
```

Comment:
Whether we end up using show_warning, warnings.warn, or logging.log (could we hook up log warnings and errors to our notifications?), here is my suggestion for the message:

> Polygons must have three vertices. Lasso polygons are simplified using an algorithm called RDP, which may cause polygons smaller than the *RDP epsilon parameter* to disappear entirely. You can change this parameter in napari > Settings > Experimental > RDP epsilon.

Maybe this is better in the docs and then we can just link to the docs:

> Simplified polygon has too few vertices. See [link] for details.